### PR TITLE
Add documentation for file-ignored and import-private-name

### PR DESCRIPTION
**Pull request by ollie-iterators**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Adds documentation for file-ignored and import-private-name

Refs #5953 

Closes #XXXX

### CODE REVIEW

File: doc/data/messages/f/file-ignored/details.rst [hunk 1]
```diff
@@ -1 +1 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !
+People can cause pylint to ignore all files by adding the code "# pylint: skip-file" to the beginning of the module
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/f/file-ignored/details.rst:1]: Shouldn't it work with Bad/good though ?

File: doc/data/messages/i/import-private-name/bad.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from classes import _PrivateClassA, safe_get_A  # [import-private-name]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/import-private-name/bad.py:1]: We could use pytest there's a lot of import. We can also show that importing private import for typing purpose is ok.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/data/messages/f/file-ignored/details.rst (hunk 1 above)
Comment (ollie-iterators) [reply doc/data/messages/f/file-ignored/details.rst:1]: So you are saying that using skip-file is bad code?

Comment (Pierre-Sassoulas) [inline doc/data/messages/f/file-ignored/details.rst:1]:
```suggestion
There's no checks at all for a file if it starts by "# pylint: skip-file".
```

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/data/messages/f/file-ignored/details.rst [hunk 3]
```diff
@@ -1 +1 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !
+There's no checks at all for a file if it starts by "# pylint: skip-file".
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/f/file-ignored/details.rst:1]:
```suggestion
There's no checks at all for a file if it starts by ``# pylint: skip-file``.
```

File: doc/data/messages/i/import-private-name/bad.py (hunk 2 above)
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/import-private-name/bad.py:1]:
```suggestion
from argparse import _SubParsersAction, _AttributeHolder  # [import-private-name]

attr_holder = _AttributeHolder()

def add_sub_parser(sub_parsers: _SubParsersAction):
    sub_parsers.add_parser('my_subparser')
    # ... 
```

File: doc/data/messages/i/import-private-name/good.py [hunk 4]
```diff
@@ -1 +1,4 @@
-# This is a placeholder for correct code for this message.
+# No error since imports are used as type annotations
+from classes import _PrivateClassA, safe_get_A
+
+a_var: _PrivateClassA = safe_get_A()
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/import-private-name/good.py:4]:
```suggestion
"""Private import can be used as type annotations."""

from argparse import _SubParsersAction

def add_sub_parser(sub_parsers: _SubParsersAction):
    sub_parsers.add_parser('my_subparser')
    # ... 
```
```

File: doc/data/messages/i/import-private-name/details.rst [hunk 5]
```diff
@@ -1 +0,0 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/import-private-name/details.rst:1]: Using private import expose you to unexpected breaking changes in any version bump of your dependencies;
even patch versions.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**